ocaml/networkd/lib/network_config.ml

```diff
@@ -34,6 +34,51 @@ let bridge_naming_convention (device : string) =
   else
     "br" ^ device
 
+let get_list_from ~sep ~key args =
+  List.assoc_opt key args
+  |> Option.map (fun v -> Astring.String.cuts ~empty:false ~sep v)
+  |> Option.value ~default:[]
+
+let parse_ipv4_config args = function
+  | Some "static" ->
+      let ip = List.assoc "IP" args |> Unix.inet_addr_of_string in
+      let prefixlen = List.assoc "NETMASK" args |> netmask_to_prefixlen in
+      let gateway =
+        Option.map Unix.inet_addr_of_string (List.assoc_opt "GATEWAY" args)
+      in
+      (Static4 [(ip, prefixlen)], gateway)
+  | Some "dhcp" ->
+      (DHCP4, None)
+  | _ ->
+      (None4, None)
+
+let parse_ipv6_config args = function
+  | Some "static" ->
+      let ipv6_arg = List.assoc "IPv6" args in
+      let ip, prefixlen =
+        Scanf.sscanf ipv6_arg "%s@/%d" (fun ip prefixlen ->
```

Comment:
This can raise an exception if the format is not found. `Stdlib.Scanf.Scan_failure`; does this need to be handled? IT would be an internal error; I assume.

Reply:
it will be caught one level up here:
```
  with e ->
    error "Error while trying to read firstboot data: %s\n%s"
      (Printexc.to_string e)
      (Printexc.get_backtrace ()) ;
    raise Read_error

```